relay.go

```diff
@@ -582,13 +584,13 @@ func (r *Relayer) handleNonCallReq(f *Frame) error {
 	sent, failure := item.destination.Receive(f, frameType)
 	if !sent {
 		r.failRelayItem(items, originalID, failure, errFrameNotSent)
-		return nil
+		return _relayShouldRelease, nil
 	}
 
 	if finished {
 		r.finishRelayItem(items, originalID)
 	}
-	return nil
+	return !sent, nil
```

Comment:
we handle the `!sent` case earlier, so this must be false at this point, so can simplify and make it explicit.

if we got to this point, the frame was sent to the other side, so no need to release.

```suggestion
	return _relayNoRelease, nil
```